### DISCUSSION TITLE
How to customise tailwind breakpoints from the downloaded CDN?

### DISCUSSION THREAD

**Discussion by shar27** [tailwindlabs/tailwindcss#9327] [Help]
From the documentation, it seems that you need to extend the theme from a tailwind.config script. However, I actually downloaded the CDN and I am serving it as a local stylesheet. I don't see a way to create or edit the responsive breakpoints. Where should I be looking?

<img width="688" alt="tailwind" src="https://user-images.githubusercontent.com/81372629/190111824-e6a82be3-9d9e-463d-b616-3bd4ce2c0752.png">

**Comment by wongjn** [comment]
You can no longer customize any Tailwind config after downloading it. This is because the Tailwind config is read **before** the stylesheet is generated – you have downloaded the **result** of this build. If you look at the response of the CDN, (https://cdn.tailwindcss.com/), you'll see that it is actually JavaScript, and it is this code that reads any Tailwind config before generating the CSS.

What you have now is any old stylesheet, that works like any other, so to edit the media queries, you can manually edit the file for your desired breakpoints.

**Reply by shar27** [reply]
Please show me an example of what you mean. Thanks.

**Reply by wongjn** [reply]
Using your screenshot as an example, we can edit the breakpoint on line 272 of `tailwind.css`:

1. Open `tailwind.css`.
2. Navigate to line 272.
3. Highlight/select the character sequence `640`.
4. Hit the <kbd>delete</kbd> or <kbd>backspace</kbd> key on your keyboard.
5. Type <kbd>5</kbd><kbd>0</kbd><kbd>0</kbd>
6. Save the file.

With the above, we have changed the media query for the `.container` from `640px` minimum width to `500px` minimum width:

```diff
 .container {
   width: 100%;
 }
-@media (min-width: 640px) {
+@media (min-width: 500px) {
   .container {
     max-width: 640px;
   }
 }
```

You can use this process to change other breakpoints in the CSS file as you wish.

**Reply by shar27** [reply]
Yes, but when you use a different example. What defines sm/md/lg?
<img width="515" alt="md mx32" src="https://user-images.githubusercontent.com/81372629/190147803-1740c53f-715e-4177-9a6b-03c71a9d1ec4.png">
<img width="605" alt="lg mx32" src="https://user-images.githubusercontent.com/81372629/190147824-f5d0ffb4-24c9-46b6-a3ed-7060c47d3603.png">

<img width="685" alt="mx issue" src="https://user-images.githubusercontent.com/81372629/190147593-9e6b321a-accf-4834-ae27-8142920c5859.png">

**Reply by wongjn** [reply]
This would have been the resolved Tailwind config at build time, which no longer exists now that you have the compiled CSS file. You should be able to scroll up or use the VSCode breadcrumb to select the `@media` wrapper for all these classes and change the media query to your taste.

**Reply by shar27** [reply]
So, basically, it's impossible to define where the actual breakpoint is.

**Reply by wongjn** [reply]
No, it is possible – it is the `@media` at-rule that is wrapping these styles. You can change this and it will redefine the breakpoint.

**Reply by shar27** [reply]
Could you show me an example please?

**Reply by wongjn** [reply]
Using your screenshot as an example:
![image](https://user-images.githubusercontent.com/81372629/190147803-1740c53f-715e-4177-9a6b-03c71a9d1ec4.png)

We can edit the breakpoint for this `md` group:

1. Scroll up in the file, until we encounter the line with `@media (min-width: 768px) {`
2. Highlight/select the character sequence `768`.
3. Hit the <kbd>delete</kbd> or <kbd>backspace</kbd> key on your keyboard.
4. Type <kbd>6</kbd><kbd>0</kbd><kbd>0</kbd>.
5. Save the file.

With the above, we have changed the media query for the `md` from `768px` minimum width to `600px` minimum width:

```diff
-@media (min-width: 768px) {
+@media (min-width: 600px) {
```
